bd-log-primitives/src/lib.rs

```diff
@@ -112,67 +114,48 @@ pub type LogFieldValue = StringOrBytes<String, Vec<u8>>;
 //
 
 /// The list of log fields annotated with extra information.
-pub type AnnotatedLogFields = Vec<AnnotatedLogField>;
+pub type AnnotatedLogFields = HashMap<String, AnnotatedLogField>;
```

Comment:
minor item that we can do as a follow up. We should see if we are already bringing any of the more performant hashers vs. the one the stdlib uses by default which is HashDoS resistant which we don't need. It would be nice to speed this up with a more performant hasher. Since we are already using a type alias for this we can always swap that in later, might as put a TODO in.

Reply:
Switched out to ahash and it didn't blow up the binary size, so that seems like a no-brainer 

Binary size does seem to have creeped up so maybe I will look at that separately